instrumentation/jaxrs/jaxrs-1.0/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v1_0/JaxRsAnnotationsTracer.java

```diff
@@ -26,23 +27,24 @@
 
   public Span startSpan(Class<?> target, Method method) {
     String pathBasedSpanName = getPathSpanName(target, method);
-    Span serverSpan = BaseTracer.getCurrentServerSpan();
+    Context context = Context.current();
+    Span serverSpan = BaseTracer.getCurrentServerSpan(context);
 
     // When jax-rs is the root, we want to name using the path, otherwise use the class/method.
     String spanName;
     if (serverSpan == null) {
       spanName = pathBasedSpanName;
     } else {
       spanName = spanNameForMethod(target, method);
-      updateServerSpanName(serverSpan, pathBasedSpanName);
+      updateServerSpanName(context, serverSpan, pathBasedSpanName);
     }
 
     return tracer.spanBuilder(spanName).startSpan();
   }
 
-  private void updateServerSpanName(Span span, String spanName) {
+  private void updateServerSpanName(Context context, Span span, String spanName) {
     if (!spanName.isEmpty()) {
-      span.updateName(spanName);
+      span.updateName(BaseTracer.getApplicationRoot(context) + spanName);
```

Comment:
How can this work? I see in JAX-RS 1.0 tests that we expect span name to be `POST /test/hello/{name}`. And now you just prepend application root to it?

Reply:
Oops! I fixed the JAX-RS 1.0 instrumentation. Still no test for JAX-RS 1.0 + non-root context path though, I added comment about this below.